blocking_queue.cpp

```diff
@@ -0,0 +1,72 @@
+#include "queue.h"
+#include "blockingconcurrentqueue.h"
+
+typedef moodycamel::BlockingConcurrentQueue<void*> BCQType, *BCQPtr;
+
+#define TO_OBJECT(x) reinterpret_cast<BCQPtr>((*x))
+
+extern "C" {
+	
+int bcq_create(BCQHandle* handle)
+{
+	if (handle == nullptr) {
+		return CQ_BAD_HANDLE;
+	}
+	try {
+		BCQPtr retval = new BCQType;
+		*handle = retval;
+	} catch(...) {
+		return CQ_ERR;
+	}
+	return CQ_OK;
+}
+
+int bcq_destroy(BCQHandle* handle)
+{
+	if (handle == nullptr) {
+		return CQ_BAD_HANDLE;
+	}
+	try {
+		BCQPtr retval = TO_OBJECT(handle);
+		delete retval;
+	} catch(...) {
+		return CQ_ERR;
+	}
+	return CQ_OK;
+}
+
+int bcq_enqueue(BCQHandle* handle, CQValue* value)
+{
+	if (handle == nullptr) {
+		return CQ_BAD_HANDLE;
+	}
+	if (value == nullptr) {
+		return CQ_BAD_VALUE_HANDLE;
+	}
+	try {
+		BCQPtr retval = TO_OBJECT(handle);
+		retval->enqueue(*value);
+	} catch(...) {
+		return CQ_ERR;
+	}
+	return CQ_OK;
+}
+
+int bcq_dequeue(BCQHandle* handle, CQValue* value)
```

Comment:
I think it would be clearer to name this after the C++ API (`bcq_wait_dequeue`)

Reply:
are you okay with the cq_ and bcq_ prefixes?

Reply:
I'd prefer `moodycamel_cq_` and `moodycamel_bcq_`. We don't want name collisions in the global namespace.

Reply:
also should it be cq_try_dequeue